src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/player/ModuleFreeCam.kt

```diff
@@ -24,83 +24,87 @@ object ModuleFreeCam : Module("FreeCam", Category.PLAYER) {
+        posY = player.y
+        posZ = player.z
+        ground = player.isOnGround
+        val faker = OtherClientPlayerEntity(world, player.gameProfile)
 
-            faker.headYaw = player.headYaw
-            faker.copyPositionAndRotation(player)
-            world.addEntity(faker.entityId, faker)
-            fakePlayer = faker
+        faker.headYaw = player.headYaw
+        faker.copyPositionAndRotation(player)
+        world.addEntity(faker.entityId, faker)
+        fakePlayer = faker
 
-            if (!collision) {
-                player.noClip = true
-            }
+        if (!collision) {
+            player.noClip = true
         }
     }
 
     override fun disable() {
-        if (mc.player == null || mc.world == null) {
-            return
-        } else {
-            player.updatePositionAndAngles(fakePlayer!!.x, fakePlayer!!.y, fakePlayer!!.z, player.yaw, player.pitch)
-            world.removeEntity(fakePlayer!!.entityId)
-            fakePlayer = null
-            player.setVelocity(x, y, z)
-        }
+        player.updatePositionAndAngles(fakePlayer!!.x, fakePlayer!!.y, fakePlayer!!.z, player.yaw, player.pitch)
+        world.removeEntity(fakePlayer!!.entityId)
+        fakePlayer = null
+        player.setVelocity(x, y, z)
     }
 
     val repeatable = repeatable {
-        if (mc.world == null) {
-            enabled = false
-        } else {
-            // Just to make sure it stays enabled
-            if (!collision) {
-                player.noClip = true
-                player.fallDistance = 0f
-                player.isOnGround = false
-            }
+        // Just to make sure it stays enabled
+        if (!collision) {
+            player.noClip = true
+            player.fallDistance = 0f
+            player.isOnGround = false
+        }
 
-            if (fly) {
-                val speed = speed.toDouble()
-                if (player.moving) {
-                    player.strafe(speed = speed)
-                }
+        if (fly) {
+            val speed = speed.toDouble()
+            if (player.moving) {
+                player.strafe(speed = speed)
+            }
 
-                player.velocity.y = when {
-                    mc.options.keyJump.isPressed -> speed
-                    mc.options.keySneak.isPressed -> -speed
-                    else -> 0.0
-                }
+            player.velocity.y = when {
+                mc.options.keyJump.isPressed -> speed
+                mc.options.keySneak.isPressed -> -speed
+                else -> 0.0
             }
         }
     }
 
     val packetHandler = handler<PacketEvent> { event ->
         when (val packet = event.packet) {
             // For better FreeCam detecting AntiCheats, we need to prove to them that the player's moving
-            is PlayerMoveC2SPacket.PositionOnly, is PlayerMoveC2SPacket.LookOnly, is PlayerMoveC2SPacket.Both -> {
+            is PlayerMoveC2SPacket -> {
                 if (spoofMovement) {
```

Comment:
I would merge this, but the `spoofMovement` option should be removed and considered always `true`

Reply:
Done.